File: src/Coduniverse.Domain/StarSystem.cs
using System.Collections.Generic;

namespace Coduniverse.Domain
{
    public class StarSystem : BaseEntity
    {
        public string Name { get; set; }
        public int Age { get; set; }
        public SpaceObject CenterMass { get; set; }
        public IList<SpaceObject> SpaceObjects { get; set; }
    }
}

File: src/Coduniverse.Client/src/app/models/star-system.ts
import { Spaceobject } from "./spaceobject";

export class StarSystem {
    constructor(
        public id: string,
        public spaceObjects: Spaceobject[],
        public name: string,
        public type: number,
        public typeName: string,
        public age: number,
        public centerMass: Spaceobject
    ) { }

}
